### PR TITLE
Feat/BNGP-3545-update-legal-agreement-parties-flow

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

File: packages/webapp/src/routes/land/add-landowner-individual.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import constants from '../../utils/constants.js'
+import {
+  getLegalAgreementDocumentType,
+  validateFirstLastName
+} from '../../utils/helpers.js'
+import isEmpty from 'lodash/isEmpty.js'
+
+const handlers = {
+  get: async (request, h) => {
```
Comment (soorajvdefra) [inline packages/webapp/src/routes/land/add-landowner-individual.js:9]: IN progress status missing 
processRegistrationTask(request, {
      taskTitle: 'Legal information',
      title: 'Add legal agreement details'
    }, {
      inProgressUrl: =
    })

File: packages/webapp/src/routes/land/add-landowner-organisation.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import constants from '../../utils/constants.js'
+import {
+  getLegalAgreementDocumentType
+} from '../../utils/helpers.js'
+
+const handlers = {
+  get: async (request, h) => {
```
Comment (soorajvdefra) [inline packages/webapp/src/routes/land/add-landowner-organisation.js:7]: in progress indicator missing  processRegistrationTask(request, {
      taskTitle: 'Legal information',
      title: 'Add legal agreement details'
    }, {
      inProgressUrl: 
    })

File: packages/webapp/src/routes/land/legal-agreement-lpa-list.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        href: `${constants.routes.LEGAL_AGREEMENT_LPA_REMOVE}?id=${index}`,
+        text: 'Remove'
+      }],
+      classes: 'govuk-summary-list__key govuk-!-font-weight-regular hmrc-summary-list__key'
+    },
+    class: 'govuk-summary-list__row'
+  }
+}
+const handlers = {
+  get: async (request, h) => {
```
Comment (soorajvdefra) [inline packages/webapp/src/routes/land/legal-agreement-lpa-list.js:30]: inprogress indicator missing
 processRegistrationTask(request, {
      taskTitle: 'Legal information',
      title: 'Add legal agreement details'
    }, {
      inProgressUrl: constants.routes.
    })

File: packages/webapp/src/routes/land/legal-agreement-lpa-list.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        routes: constants.routes,
+        err: [{
+          text: 'Select yes if you have added all landowners or leaseholders',
+          href: '#allLpa'
+        }]
+      })
+    }
+
+    if (allLpa === 'yes') {
+      return h.redirect(constants.routes.ADD_LEGAL_AGREEMENT_PARTIES)
```
Comment (soorajvdefra) [inline packages/webapp/src/routes/land/legal-agreement-lpa-list.js:64]: referrer need to setup i believ

File: packages/webapp/src/routes/land/add-landowner-individual.js (hunk 1 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/add-landowner-individual.js:10]: done

File: packages/webapp/src/routes/land/add-landowner-organisation.js (hunk 2 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/add-landowner-organisation.js:8]: done

File: packages/webapp/src/routes/land/legal-agreement-lpa-list.js (hunk 3 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/legal-agreement-lpa-list.js:34]: done

File: packages/webapp/src/routes/land/legal-agreement-lpa-list.js (hunk 4 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/legal-agreement-lpa-list.js:75]: I am not sure. @paulsimonandrews - could you confirm please? I am actually not sure what that referrer check is !?

**Review by teddmason — APPROVED** [review]

**Pull request merged**